src/mqt/predictor/driver.py

```diff
@@ -1,611 +1,13 @@
-        non_zero_indices = np.load(str(path), allow_pickle=True)
-        feature_vector = [feature_vector[i] for i in non_zero_indices]
-
-        return self.clf.predict([feature_vector])[0]
-
-    def compile_predicted_compilation_path(
-        self, qasm_str_or_path: str, prediction: int
-    ):
-        """Returns the compiled quantum circuit as a qasm string when the original qasm circuit is provided as either
-        a string or a file path and the prediction index is given."""
-
-        LUT = utils.get_index_to_comppath_LUT()
-        if prediction < 0 or prediction >= len(LUT):
-            print("Provided prediction is faulty.")
-            return None
-
-        if Path(qasm_str_or_path).exists():
-            print("Reading from .qasm path: ", qasm_str_or_path)
-            qc = QuantumCircuit.from_qasm_file(qasm_str_or_path)
-        elif QuantumCircuit.from_qasm_str(qasm_str_or_path):
-            print("Reading from .qasm str")
-            qc = QuantumCircuit.from_qasm_str(qasm_str_or_path)
-        else:
-            print("Neither a qasm file path nor a qasm str has been provided.")
-            return False
-
-        prediction_information = LUT.get(prediction)
-        gate_set_name = prediction_information[0]
-        device = prediction_information[1]
-        compiler = prediction_information[2]
-        compiler_settings = prediction_information[3]
-
-        print("")
-        if compiler == "qiskit":
-            compiled_qc = qiskit_helper.get_mapped_level(
-                qc, gate_set_name, qc.num_qubits, device, compiler_settings, False, True
-            )
-            return compiled_qc.qasm()
-        elif compiler == "tket":
-            compiled_qc = tket_helper.get_mapped_level(
-                qc,
-                gate_set_name,
-                qc.num_qubits,
-                device,
-                compiler_settings,
-                False,
-                True,
-            )
-            return circuit_to_qasm_str(compiled_qc)
-        else:
-            print("Error: Compiler not found.")
-            return False
-
-
-if __name__ == "__main__":
-
-    parser = argparse.ArgumentParser(description="Create Training Data")
-
-    parser.add_argument("--timeout", type=int, default=120)
-
-    args = parser.parse_args()
-
-    predictor = Predictor()
-
-    # Generate compiled circuits and save them as qasm files
-    predictor.generate_compiled_circuits(
-        timeout=args.timeout,
-    )
-    # Postprocess some of those qasm files
-    utils.postprocess_ocr_qasm_files()
-    # Generate training data from qasm files
-    res = predictor.generate_trainingdata_from_qasm_files()
-    # Save those training data for faster re-processing
-    utils.save_training_data(res)
-    # Train the Random Forest Classifier on created training data
-    predictor.train_random_forest_classifier()
+from mqt.predictor.ml import helper
+from mqt.predictor.ml.Predictor import Predictor
```

Comment:
knowing where this is going with the rl part, I'd rather propose that this should be
```python
from mqt.predictor import ml
```
and then use `ml.<...>`. Otherwise you might get a class of modules.